### ISSUE TITLE
Bug: player-weather issues

### ISSUE THREAD

**Issue by jaylawl** [SkriptLang/Skript#1411]
it appears to think i'm trying to change the world's weather
these issues only appear within functions apparently
```
function jgolf_setWeather(player: player, weather: weather):
    if {_weather} = rain:
        set {_player}'s weather to rain
    else if {_weather} = clear:
        set {_player}'s weather to clear
        wait 10 seconds
        reset {_player}'s weather
```
https://snag.gy/0xiWGd.jpg

**Comment by Nicofisi** [comment]
Probably it's about variables, not functions, hmm. I will look at #1408 tomorrow but I have no idea about this one, another dev needs to suggest a fix please <3

**Comment by Pikachu920** [comment]
hmmm, it's matching ExprWeather (you can tell from string form in the error). @Nicofisi can you check if your expression is ever init? if it isn't that would be bad

**Comment by ShaneBeee** [comment]
Marking this as completed as it seems it was either fixed in #2552 or maybe fixed earlier.
